src/functions/getRegeneratedEmails/framework/handler.ts

```diff
@@ -0,0 +1,36 @@
+import { APIGatewayEvent, Context } from 'aws-lambda';
+
+import { bootstrapConfig } from '../../../common/framework/config/config';
+import Response from '../../../common/application/api/Response';
+import { getRegeneratedEmails } from './repositories/get-regenerated-emails-repository';
+import { HttpStatus } from '../../../common/application/api/HttpStatus';
+import createResponse from '../../../common/application/utils/createResponse';
+import * as joi from 'joi';
+import { gzipSync } from 'zlib';
+import { RegeneratedEmailsRecord } from '../../../common/domain/regenerated-emails';
+
+export async function handler(event: APIGatewayEvent, fnCtx: Context): Promise<Response> {
+  await bootstrapConfig();
+  try {
+    const appRefPathParam = parseInt(event?.pathParameters['appRef'], 10);
+
+    const parametersSchema = joi.number().max(1000000000000);
+    const validationResult = parametersSchema.validate(appRefPathParam);
+
+    if (validationResult.error) {
+      return createResponse(validationResult.error, HttpStatus.BAD_REQUEST);
+    }
+
+    const results: RegeneratedEmailsRecord = await getRegeneratedEmails(appRefPathParam);
+    const result: RegeneratedEmailsRecord = { ...results[0] };
```

Comment:
Does `results` only ever return one object?

Reply:
yes, were searching for a app ref and aggregating all records that get returned into a single json object within the sql using the app ref as the group.
eg.
app ref: 111, details: [match 1, match 2, match3 etc].